src/libs/actions/PersonalDetails/index.native.js

```diff
@@ -0,0 +1,329 @@
+import _ from 'underscore';
+import {Platform} from 'react-native';
+import lodashGet from 'lodash/get';
+import lodashMerge from 'lodash/merge';
+import Onyx from 'react-native-onyx';
+import Str from 'expensify-common/lib/str';
+import {check, PERMISSIONS, RESULTS} from 'react-native-permissions';
+import Geolocation from 'react-native-geolocation-service';
+import ONYXKEYS from '../../../ONYXKEYS';
+import md5 from '../../md5';
+import CONST from '../../../CONST';
+import NetworkConnection from '../../NetworkConnection';
+import * as API from '../../API';
+import NameValuePair from '../NameValuePair';
+
+let currentUserEmail = '';
+Onyx.connect({
+    key: ONYXKEYS.SESSION,
+    callback: val => currentUserEmail = val ? val.email : '',
+});
+
+let personalDetails;
+Onyx.connect({
+    key: ONYXKEYS.PERSONAL_DETAILS,
+    callback: val => personalDetails = val,
+});
+
+/**
+ * Helper method to return a default avatar
+ *
+ * @param {String} [login]
+ * @returns {String}
+ */
+function getDefaultAvatar(login = '') {
+    // There are 8 possible default avatars, so we choose which one this user has based
+    // on a simple hash of their login (which is converted from HEX to INT)
+    const loginHashBucket = (parseInt(md5(login).substring(0, 4), 16) % 8) + 1;
+    return `${CONST.CLOUDFRONT_URL}/images/avatars/avatar_${loginHashBucket}.png`;
+}
```

Comment:
I'm still a bit confused as to why these functions need to be in the native only file?

For example: `getDefaultAvatar`, `getAvatar`, `getDisplayName`, etc are not unique to native. If this is a copy then we now have to maintain duplicate functions which could get a bit confusing. Is there another way to do this?

Reply:
+1